ui/src/components/layout/FlowCanvas.tsx

```diff
@@ -94,14 +101,13 @@ const FlowCanvas: React.FC<Props> = () => {
           position.y = changes[0].position.y
         } else {
           /**
-           * when dragging == false then we can use the last save position 
+           * when dragging == false then we can use the last save position
            * and set that to the DisplayNode.position property.
            * Then persist to DHT.
           */
           const store = getDataStore();
-          const planId = store.getRoot()['planId'];
-          const displayNodes: Array<DisplayNode> = store.getDisplayNodes(planId);
-          const nodeToUpdate = displayNodes.find((obj) => obj.id === changes[0].id);
```

Comment:
We can just use the cursor to get the correct node to update without having to iterate through all the nodes.